src/main/scala/uk/gov/nationalarchives/tdr/api/service/FileService.scala

```diff
@@ -42,37 +35,38 @@ class FileService(
     val allFileNodes: Map[String, TreeNode] = treeNodesUtils.generateNodes(filePaths, fileTypeIdentifier)
     val allEmptyDirectoryNodes: Map[String, TreeNode] = treeNodesUtils.generateNodes(addFileAndMetadataInput.emptyDirectories.toSet, directoryTypeIdentifier)
 
-    val folderIdToPath: Map[UUID, String] = (allEmptyDirectoryNodes ++ allFileNodes.filter(_._2.treeNodeType == directoryTypeIdentifier))
-      .map(f => f._2.id -> f._1)
+    val row: (UUID, String, String) => FilemetadataRow = FilemetadataRow(uuidSource.uuid, _, _, now, userId, _)
 
-    val fileRows: List[FileRow] = ((allEmptyDirectoryNodes ++ allFileNodes) map {
-      case (_, treeNode) =>
+    val rowsWithMatchId: List[Rows] = ((allEmptyDirectoryNodes ++ allFileNodes) map {
+      case (path, treeNode) =>
         val parentId = treeNode.parentPath.map(v => allFileNodes.getOrElse(v, allEmptyDirectoryNodes(v)).id)
```

Comment:
Maybe change the `v` param name to `path` just to make a bit clearer?

Reply:
Makes sense, I've changed it.